scalameta/src/main/scala/scala/meta/internal/parsers/Parsers.scala

```diff
@@ -1682,10 +1708,23 @@ private[meta] class Parser(val input: Input)(implicit val dialect: Dialect) { pa
    *  }}}
    */
   def caseClauses(): List[Case] = {
-    val cases = caseSeparated { caseClause() }
+    val cases = new ListBuffer[Case]
+    while (token.is[CaseIntro] || token.is[Ellipsis]) {
+      if (token.is[Ellipsis]) {
+        cases += ellipsis(1, unquote[Case], accept[`case`])
+        if (token.is[StatSep]) next()
+      } else if (token.is[`case`] && ahead(token.is[Unquote])) {
+        next()
+        cases += unquote[Case]
+        if (token.is[StatSep]) next()
```

Comment:
1) I thought about that, but it would require some extra changes in code (for cases when scala supports multiple `;`, but this parser doesn't) just for nothing. `;;;;;` is useless action.
2) Because new line might happen inside cases.